filebeat/prospector/prospector_log.go

```diff
@@ -179,6 +183,7 @@ func (p *ProspectorLog) scan() {
 		// Decides if previous state exists
 		if lastState.IsEmpty() {
 			logp.Debug("prospector", "Start harvester for new file: %s", newState.Source)
+			p.Prospector.states.Update(newState)
```

Comment:
New states should not be persisted before the harvester is started as during the startup of a harvester also some checks will be made and only then the state will be sent to the prospector.